scripts/friendly-caterpillar.py

```diff
@@ -0,0 +1,41 @@
+#! /usr/bin/python
+
+# a friendly caterpillar says hi (and other things). run as
+#   $ python friendly-caterpillar.py
+# or, if this file has executable permission,
+#   $ ./friendly-caterpillar.py
+
+import sys, time, random
+
+def scoot_back(n):
+    sys.stdout.write('{:c}{:c}{:c}'.format(8, 32, 8) * n)
```

Comment:
there was a bug here (besides the caterpillar) in my first commit. the backspace ascii character (character 8) doesn't necessarily erase as it backs up. now the line backs up, writes a space (thus moving forward while overwriting what was there), then backs up again.